File: 01_big_O_notation/03_cubic_cost/cubic_cost.js
// O(n³)

function cubicCost(n) {

  for (let i = 0; i <= n; i++) {
    console.log(i)

    for (let j = 0; j <= n; j++) {
      console.log(j)

      for (let k = 0; k <= n; k++) {
        console.log(k)
      }
    }
  }
}

function cubicCostWay02(n) {

  for (let i = 0; i <= n ** 3; i++) {
    console.log(i)
  }
}

cubicCost(3)
// cubicCostWay02(3)

File: 01_big_O_notation/03_cubic_cost/cubic_cost.py
"""
O(n³)
"""


def cubic_cost(n: int) -> None:
    for i in range(n):
        print(i)

        for j in range(n):
            print(j)

            for k in range(n):
                print(k)


def cubic_cost_way_02(n: int) -> None:
    for i in range(n ** 3):
        print(i)


if __name__ == "__main__":
    cubic_cost(n=3)
    cubic_cost_way_02(n=3)
